Source/Core/CesiumTerrainProvider.js

```diff
@@ -159,17 +154,27 @@ define([
         this._ready = false;
         this._readyPromise = when.defer();
 
-        var lastResource = resource;
-        var metadataResource = lastResource.getDerivedResource({
-            url: 'layer.json'
-        });
-
         var that = this;
+        var lastResource;
+        var metadataResource;
         var metadataError;
 
         var layers = this._layers = [];
         var attribution = '';
         var overallAvailability = [];
+        when(options.url)
```

Comment:
If options.url rejects, this will fail silently, don't we need to move this entire block into `requestMetadata` itself?  Also, request Metadata is using old (improper) promise usage, change the success and failure functions off of this promise when you move it.

Reply:
What do you want the behavior to be when `options.url` rejects?  I don't think calling `metadataFailure` is the right option because it will still try to load the tileset even if fetching the metadata fails.  And we literally can't to anything with the URL.

Reply:
Good point, I assume we want to just propagate the `url` rejection to the `readyPromise` so that it rejects.  This way a user gets the exact error.